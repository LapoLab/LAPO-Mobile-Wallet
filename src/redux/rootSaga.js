import { all } from 'redux-saga/effects';
import commonsSagas from '../commons/sagas';
import recoveryWordsSagas from '../modules/RecoveryWords/sagas';
import sendSagas from '../modules/Send/sagas';
import paperWordsConfirm from '../modules/PaperWordsConfirm/sagas';
import dashboard from '../modules/Dashboard/sagas';
import settings from '../modules/Settings/sagas';
import paperWords from '../modules/PaperWords/sagas';
import transactionsDetails from '../modules/TransactionDetails/sagas';
import chooseContact from '../modules/ChooseContact/sagas';
import splashScreen from '../modules/SplashScreen/sagas';
import switchContacts from '../modules/SwitchContacts/sagas';
import receive from '../modules/Receive/sagas';

// eslint-disable-next-line
const getListeners = (...args) => args.reduce((acc, nextArg) => [...acc, ...Object.values(nextArg).map(func => func())], []);

export default function* rootSaga() {
  yield all(
    getListeners(
      commonsSagas,
      recoveryWordsSagas,
      sendSagas,
      paperWordsConfirm,
      dashboard,
      settings,
      paperWords,
      transactionsDetails,
      chooseContact,
      splashScreen,
      switchContacts,
      receive,
    ),
  );
}
