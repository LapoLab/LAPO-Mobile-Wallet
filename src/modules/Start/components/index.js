/**
 * @flow
 */

import React, { Component } from 'react';
import {
  Text,
  View,
  TouchableOpacity,
  Image,
  ImageBackground,
  SafeAreaView,
  Linking,
  BackHandler,
} from 'react-native';
import { NavigationActions } from 'react-navigation';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  setConfirmPins as setConfirmPinsAction,
  setPins as setPinsAction,
  createWallet as createWalletAction,
  setIsRecovery as setIsRecoveryAction,
  closeApp as closeAppAction,
} from '../../../commons/actions';
import { Logo, BackgoundImage } from '../../../assets';
import styles from './styles';

type Props = {
  navigation: {
    dispatch: Function,
  },
  setConfirmPins: Function,
  setPins: Function,
  setIsRecovery: Function,
  closeApp: Function,
};
class Start extends Component<Props> {
  componentDidMount() {
    BackHandler.addEventListener('hardwareBackPress', this.handleBack);
  }

  componentWillUnmount() {
    BackHandler.removeEventListener('hardwareBackPress', this.handleBack);
  }

  handleBack = async () => {
    const { closeApp } = this.props;
    closeApp();
    return true;
  };

  navigateToScreen = route => () => {
    const {
      navigation: { dispatch },
    } = this.props;
    const navigateAction = NavigationActions.navigate({
      routeName: route,
    });
    dispatch(navigateAction);
  };

  handleCreateWallet = async () => {
    const { setConfirmPins, setPins, setIsRecovery } = this.props;
    setConfirmPins('');
    setPins('');
    setIsRecovery(false);

    this.navigateToScreen('SetPins')();
  };

  handleRecoveryWallet = () => {
    const { setConfirmPins, setPins, setIsRecovery } = this.props;
    setConfirmPins('');
    setPins('');
    setIsRecovery(true);

    this.navigateToScreen('SetPins')();
  };

  handleGoToFaqClick = () => {
    const url = 'http://support.lapo.io/?sid=17858';
    Linking.canOpenURL(url).then((supported) => {
      if (supported) {
        Linking.openURL(url);
      } else {
        // eslint-disable-next-line
        console.log(`Don't know how to open URI: ${url}`);
      }
    });
  };

  render() {
    return (
      <ImageBackground style={{ flex: 1 }} source={BackgoundImage}>
        <SafeAreaView style={styles.container}>
          <View style={styles.header}>
            <Image style={styles.imgLogo} source={Logo} />
          </View>

          <View style={styles.groupButtons}>

            <TouchableOpacity style={styles.buttonMain} onPress={this.handleCreateWallet}>
              <Text style={styles.textButton}>CREATE A NEW WALLET</Text>
            </TouchableOpacity>


            <TouchableOpacity style={styles.buttonRecovery} onPress={this.handleRecoveryWallet}>
              <Text style={styles.textButton}>RECOVER AN EXISTING WALLET</Text>
            </TouchableOpacity>
          </View>

          <View style={styles.footer}>
            <Text style={styles.helpFooter}>Need any help?</Text>
            <TouchableOpacity onPress={this.handleGoToFaqClick}>
              <Text style={styles.faqFooter}>FAQ</Text>
            </TouchableOpacity>
          </View>
        </SafeAreaView>
      </ImageBackground>
    );
  }
}

const mapDispatchToProps = dispatch => bindActionCreators(
  {
    createWallet: createWalletAction,
    setConfirmPins: setConfirmPinsAction,
    setPins: setPinsAction,
    setIsRecovery: setIsRecoveryAction,
    closeApp: closeAppAction,
  },
  dispatch,
);

export default connect(
  null,
  mapDispatchToProps,
)(Start);
