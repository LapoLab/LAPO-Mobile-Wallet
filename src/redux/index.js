import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import storage from 'redux-persist/lib/storage';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import rootReducer from './rootReducer';
import rootSagas from './rootSaga';

const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
  blacklist: [
    'modals',
    'dashboard',
    'paperWords',
    'chooseContact',
    'send',
    'transactionsDetails',
    'alert',
    'pinsRecovery',
  ],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const sagaMiddleware = createSagaMiddleware();
// eslint-disable-next-line no-undef
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(
  persistedReducer,
  composeEnhancers(applyMiddleware(sagaMiddleware)),
);
sagaMiddleware.run(rootSagas);

export const persistor = persistStore(store);
