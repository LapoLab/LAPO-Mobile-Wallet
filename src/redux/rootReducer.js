import { combineReducers } from 'redux';

import wallet from '../commons/reducers/wallet';
import pins from '../commons/reducers/pins';
import pinsRecovery from '../commons/reducers/pinsRecovery';
import modals from '../commons/reducers/modals';
import settings from '../commons/reducers/settings';
import notify from '../modules/Notify/reducers';
import paperConfirm from '../modules/PaperWordsConfirm/reducer';
import dashboard from '../modules/Dashboard/reducers';
import paperWords from '../modules/PaperWords/reducers';
import chooseContact from '../modules/ChooseContact/reducers';
import send from '../modules/Send/reducers';
import transactionsDetails from '../modules/TransactionDetails/reducer';
import alert from '../modules/Alert/reducers';

export default combineReducers({
  wallet,
  pins,
  pinsRecovery,
  notify,
  modals,
  settings,
  paperConfirm,
  dashboard,
  paperWords,
  chooseContact,
  send,
  transactionsDetails,
  alert,
});
