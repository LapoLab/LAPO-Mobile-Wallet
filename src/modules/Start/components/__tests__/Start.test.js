// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../../redux';
import Start from '../index';

describe('Start page', () => {
  test('renders snapshots', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Start />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
