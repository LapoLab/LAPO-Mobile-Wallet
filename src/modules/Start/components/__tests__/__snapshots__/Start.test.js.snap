// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start page renders snapshots 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/backgoundStart.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <RCTSafeAreaView
    style={
      Object {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "paddingTop": 60,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/logo-medium-full.png",
          }
        }
        style={
          Object {
            "height": 180,
            "resizeMode": "contain",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4183BD",
            "borderColor": "#b0cae0",
            "borderRadius": 74,
            "borderWidth": 8,
            "height": 64,
            "justifyContent": "center",
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Lato-Bold",
              "fontSize": 14,
            }
          }
        >
          CREATE A NEW WALLET
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#90929A",
            "borderColor": "#e9e9eb",
            "borderRadius": 74,
            "borderWidth": 8,
            "height": 64,
            "justifyContent": "center",
            "marginHorizontal": 25,
            "marginTop": 21,
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Lato-Bold",
              "fontSize": 14,
            }
          }
        >
          RECOVER AN EXISTING WALLET
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "padding": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(0,0,0,0.50)",
            "fontFamily": "Lato-Light",
            "fontSize": 14,
            "lineHeight": 12,
            "padding": 2,
            "paddingBottom": 5,
            "textAlign": "center",
          }
        }
      >
        Need any help?
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5D70F1",
              "fontFamily": "Lato-Regular",
              "fontSize": 14,
            }
          }
        >
          FAQ
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
